src/modules/emotes/socket-client-listeners.js

```diff
@@ -0,0 +1,55 @@
+import socketClient, {deserializeSocketChannel, EventNames} from '../../socket-client.js';
+import {getCurrentChannel} from '../../utils/channel.js';
+import twitch from '../../utils/twitch.js';
+import watcher from '../../watcher.js';
+import channelEmotes from './channel-emotes.js';
+
+function validChannelDestination(channel) {
+  const currentChannel = getCurrentChannel();
+  const [provider, providerId] = deserializeSocketChannel(channel);
+  return currentChannel.id === providerId && currentChannel.provider === provider;
+}
+
+socketClient.on(EventNames.EMOTE_CREATE, ({channel, emote}) => {
+  if (!validChannelDestination(channel)) {
+    return;
+  }
+
+  channelEmotes.upsertChannelEmote(emote);
+
+  watcher.emit('emotes.updated');
+  twitch.sendChatAdminMessage(`${emote.code} has been added`, true);
+});
+
+socketClient.on(EventNames.EMOTE_UPDATE, ({channel, ...payload}) => {
+  if (!validChannelDestination(channel)) {
+    return;
+  }
+
+  const emote = channelEmotes.getEligibleEmoteById(payload.emote.id);
+  if (emote == null) {
+    return;
+  }
+
+  channelEmotes.emotes.delete(emote.code);
+  channelEmotes.upsertChannelEmote({...emote, ...payload.emote});
+
+  watcher.emit('emotes.updated');
+});
+
+socketClient.on(EventNames.EMOTE_DELETE, ({channel, emoteId}) => {
+  if (!validChannelDestination(channel)) {
+    return;
+  }
+
+  const emote = channelEmotes.getEligibleEmoteById(emoteId);
+
+  if (emote == null) {
+    return;
+  }
+
+  channelEmotes.emotes.delete(emote.code);
+
+  watcher.emit('emotes.updated');
+  twitch.sendChatAdminMessage(`${emote.code} has been removed`, true);
```

Comment:
nit: `BetterTTV: The emote ${emote.code} has been removed`

Reply:
what about BetterTTV: ${emote.code} has been removed ?